### ISSUE TITLE
Android gamepad input not registering in menu after closing content

### ISSUE THREAD

**Issue by nutbro** [libretro/RetroArch#2827]
Please see this thread for more information:
http://libretro.com/forums/showthread.php?t=5387

logcat is here: https://gist.github.com/nutbro/fff5505212532095c4c5002c635aff6c

Interestingly, it is possible to map buttons and to create a new autoconf, but RetroArch won't translate pressing buttons into menu navigation. You are forced to use the touchscreen. Upon loading a new game, everything works as expected, remaps get loaded etc.

**Comment by fr500** [comment]
is this still happening? it certainly doesn't happen on my shield

**Comment by nutbro** [comment]
Still happening on 2016-04-12. Also tested on my Nexus 9, same problem. Both have CM13, but I don't suspect this to be the problem.
Do you use overrides? I have them like this:

https://gist.github.com/nutbro/db9f528698010dfcbcd7f8d3998abc9d

**Comment by nutbro** [comment]
I found out today how to edit keyboard layouts for Android under /system/usr/keylayout. For the record: hitting the PS key on my DS3 Controller returned me to my Nexus 5's homescreen, effectively making the key unusable for RetroArch. Commenting out the line for the PS key on Vendor_054c_Product_0268.kl made it possible to use it as a menu key. Funny enough, the stock file is the same on my Nexus 9, and there it works.

Anyway - this was when I discovered something: the bug does not occur using menu_toggle_btn from gamepad. But when using menu_toggle from overlay, something is hosed.

ingame -> input_menu_toggle_btn from gamepad -> GLUI -> "Close Content" -> everything ok
menu_toggle from overlay -> GLUI -> "Close Content" -> Gamepad input not working, touch ok

Can somebody confirm this?
I used the default retropad.

By the way: I noticed that the menu_toggle action might be faulty.
This all works:
ingame -> input_menu_toggle_btn from gamepad -> GLUI -> input_menu_toggle_btn from gamepad -> ingame
ingame -> input_menu_toggle_btn from gamepad -> GLUI -> "Resume Content" -> ingame
ingame -> menu_toggle from overlay -> GLUI -> "Resume Content" -> ingame

This not:
ingame -> menu_toggle from overlay -> GLUI -> input_menu_toggle_btn from gamepad -> nothing happens

I guess there is a problem in menu_driver.c , like events not registering.
